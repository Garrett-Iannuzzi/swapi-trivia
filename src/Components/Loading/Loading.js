import React from 'react';
import './Loading.scss';
import images from '../.././images/images.js';

const Loading = () => {
  return (
    <main className="loading-container">
      <h1>Loading...</h1>
      <section className='loading-gif-container'>
        <img className='loading-racer-gif' src={images.rider} alt='Gif of space rider' />
      </section>
    </main>
  )
}

export default Loading;
