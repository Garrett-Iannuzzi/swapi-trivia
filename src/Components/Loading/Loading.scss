.loading-container {
  height: 87vh;
  display: flex;
  flex-direction: column;
  align-items: center;
}


.loading-gif-container {
  width: 100%;
  display: flex;
  align-items: flex-end;
  height: 50vh;
}

.loading-racer-gif {
  animation: flying 4s linear infinite;
  opacity: 1;
}

@keyframes flying {
  from {
    margin-left: 95%;
    margin-right: 0;
  }
  to {
    margin-left: 0;
    margin-right: 145%;
  }
}
