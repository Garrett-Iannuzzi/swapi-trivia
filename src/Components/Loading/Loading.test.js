import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Loading from '../Loading/Loading.js';

describe('Loading', () => {

  it('Should match the snapshot with data passed in', () => {
    const wrapper = shallow(<Loading />);

    expect(wrapper).toMatchSnapshot();
  });
});
