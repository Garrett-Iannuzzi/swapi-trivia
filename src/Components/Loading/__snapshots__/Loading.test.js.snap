// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Should match the snapshot with data passed in 1`] = `
<main
  className="loading-container"
>
  <h1>
    Loading...
  </h1>
  <section
    className="loading-gif-container"
  >
    <img
      alt="Gif of space rider"
      className="loading-racer-gif"
      src="rider.gif"
    />
  </section>
</main>
`;
